### ISSUE TITLE
Automate building of all internal databases

### ISSUE THREAD

**Issue by dkoslicki** [RTXteam/RTX#1149]
Maybe using snakemake, automate the building of all internal databases. Options should include `--all`, or `--<specific database(s)>`.

**Comment by finnagin** [comment]
TODO: resurrect this issue and make sure the things that were implemented still work. Also to update to refresh databases rather than a fresh build

**Comment by dkoslicki** [comment]
Still TODO to decrease the amount of manual intervention required to refresh databases. Adding @chunyuma and @amykglen as additional databases have since been created (eg. DTD)

**Comment by dkoslicki** [comment]
6/22 AHM: as part of KG2/c, all databases are auto-built save for: NGD, COHD, DTD. NGD @amykglen's purview, COHD + DTD @chunyuma's

**Comment by saramsey** [comment]
OK to close?

**Comment by dkoslicki** [comment]
Hard to say: it would be great to have the _build_ of all databases automated, but that seems like a heavy lift.

**Comment by saramsey** [comment]
We are making progress on this, at least as concerns automating the building of KG2c downstream of KG2pre. 

See the `issue-40` branch of the "Translator-CATRAX/stitch" project area:
https://github.com/Translator-CATRAX/stitch/issues/40

**Comment by ecwood** [comment]
I agree with Steve that we have made significant progress on this issue by bringing KG2c into the KG2pre build process. However, to make this practical, we have cut aspects of the KG2c build (such as using `biolink:same_as` edges in synonymization, which is also the new Translator standard) and reworked others (Steve's `stitch` to create a local NodeNorm SQLite).
Further, in order to control costs, we build RTX-KG2 on a fairly small instance now (16 cores, 128 GiB of memory). I highly doubt this would be practical for the other databases (especially xDTD). I also doubt it would be worth the cost increase to build RTX-KG2 on a larger instance to bring the other databases into the build process. So, in addition to Steve's cost-benefit analysis, I think we need to consider instance size for this automation task.
In general, I am in favor of automating database generation when possible, though, because it reduces the rollout delays due to human hand-offs.

**Comment by saramsey** [comment]
Perhaps we should embrace the "separation of concerns" implicit in the dictum that Babel is in charge of normalization. That means we don't have to care about "same_as", right?  

1. I claim that the vast majority of `same_as` edges will be self-loops (i.e., self-edges) in the (Babel-normalized) KG2c graph, and thus, irrelevant. 
2. I claim that those few `same_as` edges that would actually not be self-loops in the KG2c graph, would be (by definition) wrong, at least insofar as Babel is the source of canonicalization/normalization truth; thus, they should be discarded.

With apologies to Dr. Strangelove, I'm entitling this post "How I learned to love the Babel".

**Comment by saramsey** [comment]
This issue is superseded by #2730
